Hestia/app/src/androidTest/java/com/rugged/application/hestia/UI/dialogs/IpDialogTest.java

```diff
@@ -1,37 +1,72 @@
 package com.rugged.application.hestia.UI.dialogs;
 
+import android.support.test.espresso.matcher.ViewMatchers;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import com.rugged.application.hestia.R;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import hestia.UI.activities.home.HomeActivity;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.clearText;
 import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.withDecorView;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
-import static junit.framework.Assert.assertEquals;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
 
-/**
 
 @RunWith(AndroidJUnit4.class)
-public class SetIpTest {
+public class IpDialogTest {
     private final String SET_IP_TEXT = "Set IP ";
-    private final String TEST_IP = "192.168.0.1";
+    private final String VALID_IP = "192.168.0.1";
+    private final String INVALID_IP = "0.3000.283.1";
+    private final String INCORR_IP_TOAST="The IP address should be between 0.0.0.0 and 255.255.255.255";
 
     @Rule
-    public IntentsTestRule<DeviceListActivity> mIntentsRule =
-            new IntentsTestRule<>(DeviceListActivity.class);
+    public ActivityTestRule<HomeActivity> activityTestRule =
+            new ActivityTestRule<>(HomeActivity.class);
 
-    @Before
-    public void checkDialog() {
+    public void openDialog() {
         onView(ViewMatchers.withId(R.id.context_menu)).perform(click());
         onView(withText(SET_IP_TEXT)).perform(click());
     }
 
     @Test
-    public void checkEnterIp(){
-        onView(withId(R.id.ip)).perform(clearText(),typeText(TEST_IP), closeSoftKeyboard());
+    public void checkValidIp(){
+        openDialog();
+
+        onView(withId(R.id.ip)).perform(clearText(),typeText(VALID_IP), closeSoftKeyboard());
 
-        onView(withId(R.id.confirm_button)).perform(click());
+        onView(withText("Confirm")).perform(click());
```

Comment:
hmm, I believe you should use the resources here. Or is there a reason not to? If the text on the buttons is hard-coded, then it should be fixed as well.